### ISSUE TITLE
MvNormal logp eval error

### ISSUE THREAD

**Issue by junpenglao** [pymc-devs/pymc#2313]
Running
```python
pm.MvNormal.dist(mu=np.array([0.,0.]), cov=np.diag([1.,1.])).logp(np.array([0.,0.])).eval()
```
Raise:
```python
Traceback (most recent call last):

  File "<ipython-input-18-d8bf3e2a76fd>", line 1, in <module>
    pm.MvNormal.dist(mu=np.array([0.,0.]), cov=np.diag([1.,1.])).logp(np.array([0.,0.])).eval()

  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/pymc3/distributions/multivariate.py", line 179, in logp
    value = value.reshape((-1, k))

TypeError: 'TensorVariable' object cannot be interpreted as an integer
```

**Comment by junpenglao** [comment]
```python
value = value.reshape((-1, k))
```
in [distributions/multivariate.py Line 179](https://github.com/pymc-devs/pymc3/blob/master/pymc3/distributions/multivariate.py#L179) might be problematic as it also give error to the following:
```python
with pm.Model() as model:
    pm.MvNormal('test', mu=np.zeros(2), cov=np.eye(2))
```
```python
Traceback (most recent call last):

  File "<ipython-input-13-5dc8c0c04417>", line 2, in <module>
    pm.MvNormal('test', mu=np.zeros(2), cov=np.eye(2))

  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/pymc3/distributions/distribution.py", line 38, in __new__
    return model.Var(name, dist, data, total_size)

  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/pymc3/model.py", line 515, in Var
    total_size=total_size, model=self)

  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/pymc3/model.py", line 869, in __init__
    self.logp_elemwiset = distribution.logp(self)

  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/pymc3/distributions/multivariate.py", line 179, in logp
    value = value.reshape((-1, k))

  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/theano/tensor/var.py", line 321, in reshape
    return theano.tensor.basic.reshape(self, shape, ndim=ndim)

  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/theano/tensor/basic.py", line 4910, in reshape
    rval = op(x, newshape)

  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/theano/gof/op.py", line 625, in __call__
    storage_map[ins] = [self._get_test_value(ins)]

  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/theano/gof/op.py", line 562, in _get_test_value
    ret = v.type.filter(v.tag.test_value)

  File "/Library/Frameworks/Python.framework/Versions/3.5/lib/python3.5/site-packages/theano/tensor/type.py", line 178, in filter
    data.shape))

TypeError: For compute_test_value, one input test value does not have the requested type.

The error when converting the test value to that variable type:
Wrong number of dimensions: expected 0, got 1 with shape (2,).
```

**Comment by aseyboldt** [comment]
I think this should be fixed by #2262.

**Comment by junpenglao** [comment]
@aseyboldt yep it is fixed in that PR.
